includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -505,6 +506,43 @@ public static function get_or_create_customer_from_order( $order ) {
 		return $result ? $customer_id : false;
 	}
 
+	/**
+	 * Try to get a customer name from user profile or order information.
+	 *
+	 * @param int      $user_id User ID.
+	 * @param WC_Order $order Order made by customer.
+	 * @return array
+	 */
+	public static function get_customer_name( $user_id = 0, $order = null ) {
+		$first_name = null;
+		$last_name  = null;
+
+		if (
+			$user_id &&
+			get_user_meta( $user_id, 'first_name', true ) ||
+			get_user_meta( $user_id, 'last_name', true )
+		) {
+			$first_name = get_user_meta( $user_id, 'first_name', true );
+			$last_name  = get_user_meta( $user_id, 'last_name', true );
+		} elseif (
+			$order &&
+			$order->get_billing_first_name( 'edit' ) ||
+			$order->get_billing_last_name( 'edit' )
+		) {
+			$first_name = $order->get_billing_first_name( 'edit' );
+			$last_name  = $order->get_billing_last_name( 'edit' );
+		} elseif (
+			$order &&
+			$order->get_shipping_first_name( 'edit' ) ||
+			$order->get_shipping_last_name( 'edit' )
+		) {
+			$first_name = $order->get_shipping_first_name( 'edit' );
+			$last_name  = $order->get_shipping_last_name( 'edit' );
+		}
+
+		return apply_filters( 'woocommerce_customer_name', array( $first_name, $last_name ), $order );
```

Comment:
Should we use "report" somewhere in this filter name?

Reply:
Sure!  Prefixed with `woocommerce_reports_` to keep consistent with other filters.